sygnal/gcmpushkin.py

```diff
@@ -482,23 +493,15 @@ def __init__(self):
             mapping of registration ID to either its canonical registration ID,
             or `None` if there is no entry.
         """
-        return {reg_id: await self.get_canonical_id(reg_id) for reg_id in reg_ids}
-
-    async def get_canonical_id(self, reg_id):
-        """
-        Retrieves the canonical registration ID for one registration ID.
-
-        Args:
-            reg_id (str): registration ID to retrieve the canonical registration
-                ID for.
-
-        Returns (dict):
-            its canonical registration ID, or `None` if there is no entry.
-        """
-        rows = await self.db.runQuery(
-            "SELECT canonical_reg_id FROM gcm_canonical_reg_id WHERE reg_id = ?",
-            (reg_id,),
+        rows = dict(
+            await self.db.runQuery(
+                """
+            SELECT reg_id, canonical_reg_id
+            FROM gcm_canonical_reg_id
+            WHERE reg_id IN (%s)
+            """
```

Comment:
this lot could do with more indentation, for consistency with other `"""` blocks